### PR TITLE
WX-1351 CI CWL cleanup

### PR DESCRIPTION
**Pull request by aednichols**
I started doing this opportunistically while working on CI speedup but it got out of hand. Easier to have a clean diff & CI run with a single purpose.

### CODE REVIEW

**Comment by THWiseman** [comment]
Appreciate you keeping these test/build scripts in tip-top shape. Big fan of red diffs!

File: src/ci/bin/test.inc.sh [hunk 1]
```diff
@@ -355,15 +355,11 @@ cromwell::private::create_build_variables() {
...
     backend_type="$(echo "${backend_type}" | sed 's/\([A-Z]\)/_\1/g' | tr '[:upper:]' '[:lower:]' | cut -c 2-)"
     CROMWELL_BUILD_BACKEND_TYPE="${backend_type}"
 
-    if [[ "${CROMWELL_BUILD_TYPE}" == conformance* ]]; then
-        CROMWELL_BUILD_SBT_ASSEMBLY_COMMAND="server/assembly centaurCwlRunner/assembly"
-    else
-        CROMWELL_BUILD_SBT_ASSEMBLY_COMMAND="assembly"
-    fi
+    # It may be possible to trim this down to e.g. `server/assembly` for some jobs
+    CROMWELL_BUILD_SBT_ASSEMBLY_COMMAND="assembly"
```
Comment (THWiseman) [inline src/ci/bin/test.inc.sh:362]: Interesting opportunity for more optimization!

**Review by THWiseman — APPROVED** [review]

File: src/ci/bin/test.inc.sh [hunk 2]
```diff
@@ -355,15 +355,11 @@ cromwell::private::create_build_variables() {
     backend_type="${backend_type#centaurWdlUpgrade}"
     backend_type="${backend_type#centaurHoricromtal}"
     backend_type="${backend_type#centaur}"
-    backend_type="${backend_type#conformance}"
     backend_type="$(echo "${backend_type}" | sed 's/\([A-Z]\)/_\1/g' | tr '[:upper:]' '[:lower:]' | cut -c 2-)"
     CROMWELL_BUILD_BACKEND_TYPE="${backend_type}"
 
-    if [[ "${CROMWELL_BUILD_TYPE}" == conformance* ]]; then
```
Comment (jgainerdewar) [inline src/ci/bin/test.inc.sh:362]: TIL that "conformance" == CWL

**Review by jgainerdewar — APPROVED** [review]

File: src/ci/bin/test.inc.sh (hunk 1 above)
Comment (aednichols) [reply src/ci/bin/test.inc.sh:362]: Yeah, it seems to cut down the JAR build from 7 to 4 minutes when all you need is `server`. But it was getting to be confusing making too many changes at once so I deferred the less-impactful changes for later.

File: src/ci/bin/test.inc.sh (hunk 2 above)
Comment (aednichols) [reply src/ci/bin/test.inc.sh:362]: Yep, the CWL folks provide [standardized test cases](https://www.commonwl.org/v1.1/) for engines to run against, which is actually a pretty great idea.

**Pull request merged**